annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+        "annotation.tasks.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+        mock_bulk_insert_mappings = MagicMock()
+        mock_session.bulk_insert_mappings = mock_bulk_insert_mappings
+        mock_session.flush = MagicMock()
+
+        create_tasks(mock_session, tasks, job_id)
+
+        assert mock_bulk_insert_mappings.call_count == 1
+        assert mock_update_files.call_count == 1
+        mock_update_user_overall_load.assert_not_called()
+
+
+def create_task(status: TaskStatusEnumSchema) -> ManualAnnotationTask:
+    task = MagicMock(spec=ManualAnnotationTask)
+    task.status = status
+    task.id = 1
+    return task
+
+
+def test_update_task_status_ready(mock_session: Mock):
+    task = create_task(TaskStatusEnumSchema.ready)
+
+    update_task_status(mock_session, task)
+
+    assert task.status == TaskStatusEnumSchema.in_progress
+    mock_session.add.assert_called_once_with(task)
+    mock_session.commit.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("status", "expected_message"),
+    (
+        (TaskStatusEnumSchema.pending, "Job is not started yet"),
+        (TaskStatusEnumSchema.finished, "Task is already finished"),
+    ),
+)
+def test_update_task_status_error(
+    mock_session: Mock, status: TaskStatusEnumSchema, expected_message: str
+):
+    task = create_task(status)
+    with pytest.raises(FieldConstraintError, match=f".*{expected_message}.*"):
+        update_task_status(mock_session, task)
+
+
+def test_finish_validation_task(mock_session: MagicMock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.with_for_update.return_value = mock_query
+    mock_query.update.return_value = None
+
+    finish_validation_task(mock_session, mock_task)
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+
+    mock_query.with_for_update.assert_called_once()
+    mock_query.update.assert_called_once_with(
+        {ManualAnnotationTask.status: TaskStatusEnumSchema.finished},
+        synchronize_session="fetch",
+    )
+    mock_session.commit.assert_called_once()
+
+
+def test_count_annotation_tasks(mock_session: Mock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.count.return_value = 5
+
+    result = count_annotation_tasks(mock_session, mock_task)
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_query.count.assert_called_once()
+
+    assert result == 5
+
+
+@pytest.fixture
+def mock_task_revisions() -> List[AnnotatedDoc]:
+    mock_revision = MagicMock(spec=AnnotatedDoc)
+    mock_revision.pages = {"1": ["data1"], "2": ["data2"]}
+    mock_revision.failed_validation_pages = [1, 2]
+    mock_revision.validated = [1, 2]
```

Comment:
While writing test for my functionality, I understood that pages can just be in one of the three sets (pages, failed, validated) exclusively. Though, I understood the need to have the same integers in all sets, for easier testing of your functions.